pkg/tests/upgrade/test_salt_upgrade.py

```diff
@@ -1,37 +1,39 @@
 import pytest
 
 
-def test_salt_upgrade(salt_call_cli, salt_minion, install_salt):
+def test_salt_upgrade(salt_call_cli, install_salt):
     """
-    Test upgrade of Salt
+    Test an upgrade of Salt.
     """
     if not install_salt.upgrade:
         pytest.skip("Not testing an upgrade, do not run")
-    # verify previous install version is setup correctly and works
-    ret = salt_call_cli.run("test.ping")
+
+    if install_salt.relenv:
+        original_py_version = install_salt.package_python_version()
+
+    # Verify previous install version is setup correctly and works
+    ret = salt_call_cli.run("test.version")
     assert ret.returncode == 0
-    assert ret.data
+    assert ret.data < install_salt.artifact_version
```

Comment:
```suggestion
    assert ret.data < install_salt.artifact_version
```
Are we comparing strings here?